budgetkey_data_pipelines/pipelines/budget/national/changes/processed/extract-change-groups.py

```diff
@@ -0,0 +1,288 @@
+                                logging.debug('%s %s %s %s' % (
+                                    date, len(date_reserve), num_found, i
+                                ))
+                            group = date_groups.send(found)
+                            found = False
+                            sumvec = sum(
+                                sum(c.get(x, 0) for c in group) ** 2
+                                for x in value_fields
+                            )
+                            if sumvec == 0:
+                                pending = group[0]['pending']
+                                transfer_codes = set(x['trcode'] for x in group)
+                                if len(selected_transfer_codes & transfer_codes) > 0:
+                                    continue
+                                selected_transfer_codes.update(transfer_codes)
+                                transfer_codes = list(transfer_codes)
+                                transfer_codes.sort()
+                                num_found += len(transfer_codes)
+                                to_append = {
+                                    'transfer_ids': transfer_codes,
+                                    'date': date,
+                                    'pending': pending,
+                                }
+                                result.append(to_append)
+                                found = True
+                    except StopIteration:
+                        pass
+        for change in changes:
+            if not change['trcode'] in selected_transfer_codes:
+                selected_transfer_codes.add(change['trcode'])
+                result.append({
+                    'transfer_ids': [change['trcode']],
+                    'date': change['date'],
+                    'pending': change['pending']
+                })
+        return result
+
+    def build_groups(groups, changes):
+        for group in groups:
+            trcodes = set(group['transfer_ids'])
+            years = set(
+                int(x.split('/')[0])
+                for x in trcodes
+            )
+            assert(len(years) == 1)
+
+            group['transfer_ids'] = list(set(
+                x.split('/')[1]
+                for x in trcodes
+            ))
+            group['group_id'] = group['transfer_ids'][0]
+
+            transfer_changes = list(filter(
+                lambda x: x['trcode'] in trcodes,
+                changes
+            ))
+            group['budget_codes'] = list(set(
+                x['budget_code']
+                for x in transfer_changes
+            ))
+            group['prefixes'] = list(set(
+                itertools.chain.from_iterable(
+                    [code[:l] for l in range(2, 10, 2)]
+                    for code in group['budget_codes']
+                )
+            ))
+            group['prefixes'].sort(key=lambda x: int('1' + x))
+            group['changes'] = [
```

Comment:
just adding here but relevant to all code - 
can we make sure that the terminology in the code is about requests and transactions (as described in the issue) and not about groups.
I know that this is the way it was in the original code, but I want to make things more consistent with other parts of the system.